src/TRestGeant4ToDetectorHitsProcess.cxx

```diff
@@ -202,30 +203,20 @@ TRestEvent* TRestGeant4ToDetectorHitsProcess::ProcessEvent(TRestEvent* evInput)
     fHitsEvent->SetTimeStamp(fG4Event->GetTimeStamp());
     fHitsEvent->SetState(fG4Event->isOk());
 
-    Int_t i, j;
-    Double_t x, y, z, E;
-
-    for (i = 0; i < fG4Event->GetNumberOfTracks(); i++) {
-        for (j = 0; j < fG4Event->GetTrack(i)->GetNumberOfHits(); j++) {
-            // read x,y,z and E of every hit in the G4 event
-            x = fG4Event->GetTrack(i)->GetHits()->fX[j];
-            y = fG4Event->GetTrack(i)->GetHits()->fY[j];
-            z = fG4Event->GetTrack(i)->GetHits()->fZ[j];
-            E = fG4Event->GetTrack(i)->GetHits()->fEnergy[j];
-
-            Bool_t addHit = true;
```

Comment:
Indeed, we change the behaviour, by default `addHit = true` unless the number of hits is higher than zero

Reply:
> Indeed, we change the behaviour, by default `addHit = true` unless the number of volume hits is higher than zero

I am not sure if this refers to the current code (77c74c1d35a134a01be50d10a5d1f94ab4822f2e), I think the behaviour right now is the same, right?

Reply:
No, it is not the same, before we had `addHit = true` if `fVolumeId.size() == 0` now we just check `hits.GetVolumeId(j) == volumeID)`. So, in case `fVolumeId.size() == 0` (which I would suggest `fVolumeId.empty()`) is not checked. Therefore, the  logic has been changed.

Reply:
Right, however if `fVolumeId.size() == 0`, then we would have no hits inside the event. In normal conditions, the loop will always go inside, setting `addHit=false`. Because, if there are hits it means that there was at least one `activeVolume` and `fVolumeId` should contain at least 1 volume id.

Reply:
I think [53d73ec](https://github.com/rest-for-physics/connectorslib/pull/10/commits/53d73ec8fa2cff71a8f71783fc2b3cc22a559fea) restores previous logic, perhaps now pipeline passes 🤞🏻